econml/dowhy.py

```diff
@@ -0,0 +1,222 @@
+                                                     method_params={
+                                                         "init_params": init_params,
+                                                         "fit_params": kwargs,
+                                                     },
+                                                     )
+        return self
+
+    def refute_estimate(self, *, method_name, **kwargs):
+        """Refute an estimated causal effect.
+
+        If method_name is provided, uses the provided method. In the future, we may support automatic
+        selection of suitable refutation tests.
+        Following refutation methods are supported:
+            * Adding a randomly-generated confounder: "random_common_cause"
+            * Adding a confounder that is associated with both treatment and outcome: "add_unobserved_common_cause"
+            * Replacing the treatment with a placebo (random) variable): "placebo_treatment_refuter"
+            * Removing a random subset of the data: "data_subset_refuter"
+
+        Parameters
+        ----------
+        method_name: string
+            Name of the refutation method
+        kwargs: optional
+            Additional arguments that are passed directly to the refutation method.
+            Can specify a random seed here to ensure reproducible results ('random_seed' parameter).
+            For method-specific parameters, consult the documentation for the specific method.
+            All refutation methods are in the causal_refuters subpackage.
+
+        Returns
+        -------
+        RefuteResult: an instance of the RefuteResult class
+        """
+        return self.dowhy_.refute_estimate(
+            self.identified_estimand_, self.estimate_, method_name=method_name, **kwargs
+        )
+
+    # We don't allow user to call refit_final from this class, since internally dowhy effect estimate will only update
+    # cate estimator but not the effect.
+    def refit_final(self, inference=None):
+        raise AttributeError(
+            "Method refit_final is not allowed here! Please call it from Cate Estimator directly! ")
+
+    def __getattr__(self, attr):
+        # don't proxy special methods
+        if attr.startswith('__'):
+            raise AttributeError(attr)
+        elif attr in ['_cate_estimator', 'dowhy_',
+                      'identified_estimand_', 'estimate_']:
+            self.__dict__[attr] = value
+        elif attr.startswith('dowhy__'):
+            return getattr(self.dowhy_, attr[7:])
+        elif hasattr(self.estimate_._estimator_object, attr):
+
+            return getattr(self.estimate_._estimator_object, attr)
+        else:
+            return getattr(self.dowhy_, attr)
+
+    def __setattr__(self, attr, value):
+        if attr in ['_cate_estimator', 'dowhy_',
+                    'identified_estimand_', 'estimate_']:
+            self.__dict__[attr] = value
```

Comment:
Assuming it works, I think maybe 

```suggestion
            super().__setattr__(attr, value)
```
would be slightly more understandable than setting the value directly.